pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -26,11 +23,56 @@ type notificationChannel struct {
 }
 
 // channelsPerOrg maps notification channels per organisation
-type channelsPerOrg map[int64]map[interface{}]*notificationChannel
+type channelsPerOrg map[int64][]*notificationChannel
 
 // channelMap maps notification channels per organisation
 type defaultChannelsPerOrg map[int64][]*notificationChannel
 
+// Union type for receiver map that contains both uid -> receiver and ID -> receiver
+type uidOrID interface{}
+
+// Start channel and alert migrations. This creates Alertmanager configs as well as migrates receivers and creates the initial root-level route.
+// Returns per org maps containing: The alertmanager config, all migrated receivers, all migrated default receivers
+func (m *migration) setupAlertmanagerConfigs() (amConfigsPerOrg, map[int64]map[uidOrID]*PostableApiReceiver, map[int64]map[string]struct{}, error) {
+	// allChannels: channelUID -> channelConfig
+	allChannelsPerOrg, defaultChannelsPerOrg, err := m.getNotificationChannelMap()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	amConfigPerOrg := make(amConfigsPerOrg, len(allChannelsPerOrg))
+	receiversPerOrg := make(map[int64]map[uidOrID]*PostableApiReceiver, len(allChannelsPerOrg))
+	defaultReceiversPerOrg := make(map[int64]map[string]struct{}, len(allChannelsPerOrg))
+	for orgID := range allChannelsPerOrg {
+		amConfig, ok := amConfigPerOrg[orgID]
+		if !ok {
+			amConfig = &PostableUserConfig{
+				AlertmanagerConfig: PostableApiAlertingConfig{
+					Receivers: make([]*PostableApiReceiver, 0),
+				},
+			}
+			amConfigPerOrg[orgID] = amConfig
+		}
+
+		defaultChannels := defaultChannelsPerOrg[orgID]
+		defaultReceiversPerOrg[orgID] = make(map[string]struct{})
+		for _, c := range defaultChannels {
+			defaultReceiversPerOrg[orgID][c.Name] = struct{}{}
+		}
+
+		// Create all new receivers and add to amConfig
```

Comment:
```suggestion
		// Create all new receivers and add to amConfig.
```

Reply:
nit: Use a period and capitalization for comment sentences.

I know no one wants to be told, it's probably the job of a linter and is not enforced consistently but when possible try to adhere to https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences. I have it on my agenda to write it somewhere that's the style I'd like us to follow.